models/uniswap/celo/uniswap_v3_celo_flashloans.sql

```diff
@@ -0,0 +1,52 @@
+{{ config(
+      tags = ['dunesql']
+      , alias = alias('flashloans')
+      , materialized = 'incremental'
+      , file_format = 'delta'
+      , incremental_strategy = 'merge'
+      , unique_key = ['tx_hash', 'evt_index']
+      , post_hook = '{{ expose_spells(\'["celo"]\',
+                                  "project",
+                                  "uniswap_v3",
+                                  \'["hildobby", "tomfutago"]\') }}'
+  )
+}}
+
+WITH flashloans AS (
+    SELECT f.evt_block_time AS block_time
+    , f.evt_block_number AS block_number
+    , CASE WHEN cast(f.amount0 as double) = 0 THEN f.amount1 ELSE f.amount0 END AS amount_raw
+    , f.evt_tx_hash AS tx_hash
+    , f.evt_index
+    , CASE WHEN cast(f.amount0 as double) = 0 THEN f.paid1 ELSE f.paid0 END AS fee
+    , CASE WHEN cast(f.amount0 as double) = 0 THEN p.token1 ELSE p.token0 END AS currency_contract
+    , CASE WHEN cast(f.amount0 as double) = 0 THEN erc20b.symbol ELSE erc20a.symbol END AS currency_symbol
+    , CASE WHEN cast(f.amount0 as double) = 0 THEN erc20b.decimals ELSE erc20a.decimals END AS currency_decimals
```

Comment:
can we stay consistent as other blockchain spells here and use:
`CASE WHEN f.amount0 = UINT256 '0'`

Reply:
Fixed.